Add test to validate cache behavior on layers

zoek1 opened the pull request:
Add support to three specific elements, Packages, Files and Notices and
the integration of them to ensure the correct behavior of populating
and saving given elements.

The use of mock objects in the cache test cases was implement to focus
on cache behavior and ensure one single reponsabiliby principle.

See: #539

Signed-off-by: Miguel Angel G <miguel@gordian.dev>

zoek1 commented:
Any feedback on this? :smiley_cat:

nishakm commented on tests/test_analyze_common.py:
```diff
@@ -3,10 +3,16 @@
 # Copyright (c) 2020 VMware, Inc. All Rights Reserved.
 # SPDX-License-Identifier: BSD-2-Clause
 
+from __future__ import absolute_import
```
We only support Python 3.5 and higher. Do we need this for something else?

zoek1 replied:
No, sorry. I miss delete such line. I'll remove it and update the commit message.

zoek1 replied:
Done! :smile:

nishakm commented:
@zoek1 Thanks for the PR! It mostly looks good. I asked a question in the code and a small change in the commit message below:
```
The use of mock objects in the cache test cases was implement to focus
on cache behavior and ensure one single reponsabiliby principle. --> can you elaborate on this in your commit message?
```

nishakm commented:
@zoek1 Thanks for working on this! One small nit to fix before we merge this:
```
one single reponsabiliby principle. --> please change to responsibility
```
Thanks so much!

zoek1 commented:
I updated the commit message and removed the `__future__` import. Also new changes in master broke the tests but now are fixed. I'll continue adding more tests in another PR related to `commons.py`. :smiley_cat:

nishakm commented:
@zoek1 Thanks for updating your commit message. This PR looks good. I will merge it after we cut a release (should be this week).
